x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Case.java

```diff
@@ -215,23 +224,20 @@ public boolean foldable() {
             if (condition.condition.foldable() == false) {
                 return false;
             }
-            Boolean b = (Boolean) condition.condition.fold();
-            if (b != null && b) {
+            Object o = condition.condition.fold();
+            if (o instanceof List) {
+                /*
+                 * multivalued fields fold to null which folds to false.
+                 * So they *are* foldable if the value is foldable.
+                 */
                 return condition.value.foldable();
```

Comment:
Can the list be empty? if so, is that the same as false?

Reply:
I don't think lists occur unless they have at least 2 elements.

I think the intention is to treat any multivalue the same as false, which I think is correct, but the implementation in these lines is a bug.

More interestingly, there could be lists with 1 element which is `true`. I'd say that's a bug because that should simply be a boolean literal, but I'm not sure we never create such literals, either.